src/ghost-content/index.js

```diff
@@ -266,23 +283,34 @@ module.exports = ({ logger, db, projectLocation, enabled }) => {
 
   const getPending = () => pendingRevisionsByFolder
 
-  const getPendingWithContentForFolder = async (folderInfo, normalizedFolderName) => {
+  const getPendingWithContentForFolder = ({ stringifyBinary = false }) => async (folderInfo, normalizedFolderName) => {
     const revisions = folderInfo.map(({ revision }) => revision)
     const fileNames = uniq(folderInfo.map(({ file }) => file))
+    const { isBinary } = folderOptions[normalizedFolderName]
+    const column = isBinary ? 'binary_content' : 'content'
 
     const knex = await db.get()
     const files = await knex('ghost_content')
-      .select('file', 'content', 'deleted')
+      .select('file', column, 'deleted')
       .whereIn('file', fileNames)
       .andWhere({ folder: normalizedFolderName })
 
+    if (isBinary) {
+      files.forEach(data => {
+        data.content = stringifyBinary ? data.binary_content.toString('base64') : data.binary_content
+        delete data.binary_content
+      })
+    }
+
     return {
       files,
-      revisions
+      revisions,
+      binary: isBinary
     }
```

Comment:
@emirotin , if we were to avoid mutation, we could use something like this here:

```
return {
  revisions,
  binary: isBinary,
  files: !isBinary ? files : files.map(file => ({
    ..._.omit(file, 'binary_content'),
    content: stringifyBinary ? file.binary_content.toString('base64') : file.binary_content
  }),
}
```

Reply:
Sure thing but this should be totally fine — these objects are freshly creates by knex as the representation of the DB query results. So nobody owns them but this code, we're fine to mutate them and it's just simpler in this case.